File: ee25btech11047/Matgeo/problem23/codes/12.476.c
#include <stdio.h>

#define N 3  // size of the square matrices

// Function to multiply two matrices
void multiply(int A[N][N], int B[N][N], int C[N][N]) {
    for(int i=0; i<N; i++) {
        for(int j=0; j<N; j++) {
            C[i][j] = 0;
            for(int k=0; k<N; k++)
                C[i][j] += A[i][k] * B[k][j];
        }
    }
}

// Function to print a matrix
void printMatrix(int M[N][N]) {
    for(int i=0; i<N; i++) {
        for(int j=0; j<N; j++)
            printf("%d ", M[i][j]);
        printf("\n");
    }
}

int main() {
    // Example matrices
    int A[N][N] = {{1,0,0},{0,1,0},{0,0,1}};
    int B[N][N] = {{1,0,0},{0,1,0},{0,0,1}}; // B = inverse of A (identity in this case)
    int C[N][N];

    multiply(A, B, C);

    printf("Matrix AB:\n");
    printMatrix(C);

    // Check if C is identity
    int isIdentity = 1;
    for(int i=0;i<N;i++)
        for(int j=0;j<N;j++)
            if((i==j && C[i][j]!=1) || (i!=j && C[i][j]!=0))
                isIdentity = 0;

    if(isIdentity)
        printf("Hence, B = A^-1\n");
    else
        printf("B is NOT the inverse of A\n");

    return 0;
}

File: ee25btech11047/Matgeo/problem23/codes/12.476.py
import numpy as np

# Example 3x3 matrix A
A = np.array([[2, 1, 1],
              [1, 2, 1],
              [1, 1, 2]])

# Suppose B satisfies AB = I
# Compute B = inverse of A
B = np.linalg.inv(A)

# Verify AB = I
AB = np.dot(A, B)

print("Matrix A:\n", A)
print("\nMatrix B (A^-1):\n", B)
print("\nAB:\n", AB)

# Check if AB is approximately identity
if np.allclose(AB, np.eye(3)):
    print("\nHence, B = A^-1")
else:
    print("\nB is NOT the inverse of A")
